### PR TITLE
Implementing personalization with Gorilla

### PR DESCRIPTION
**Pull request by arthbohra**
# Personalization with Bash history

Leveraging the synthesis of user's bash history  to see if we can personalize the responses from Gorilla. We also use Microsoft Presidio to anonymize their bash history as well just in case. This is to prevent the model from processing a user's private information. An example synthesis of bash history looks like:

"The user appears to be working with Python and AWS. They have used pip to install several Python packages, specifically 'presidio_analyzer' and 'presidio_anonymizer'. They have also used python commands to execute URLs. Most notably for the context of the query, user ran 'gorilla' commands relating to AWS, such as listing all files in the current directory, moving a file from one bucket to another, and listing all AWS instances. Finally, they executed an 'export' command possibly setting an AWS location. The user is now interested in their AWS history. This context suggests that the user may be doing some data analysis on AWS and wants to retrace their steps or review their activities on AWS."

### CODE REVIEW

File: go_cli.py [hunk 1]
```diff
@@ -231,14 +231,23 @@ def get_history_commands(history_file):
...
+
     if args.history:
         commands = get_history_commands(HISTORY_FILE)
     else:
         with Halo(text=f"{GORILLA_EMOJI}Loading", spinner="dots"):
             try:
                 data_json = {
                     "user_id": user_id,
-                    "user_input": user_input,
+                    "user_input": personalized_input,
```
Comment (ShishirPatil) [inline go_cli.py]: Maybe we have another field for "personalized_input"? This means I might also have to update the server side.

File: utils.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
+import os
+from presidio_analyzer import AnalyzerEngine, PatternRecognizer
+from presidio_anonymizer import AnonymizerEngine
+from presidio_anonymizer.entities import OperatorConfig
+import json
+from pprint import pprint
+from openai import OpenAI
+
+client = OpenAI(api_key=os.getenv("OPENAI_KEY"))
```
Comment (ShishirPatil) [inline utils.py]: We should have a better way to a) request permission from the user to do this, and b) give users the configuration option to change this.

File: go_cli.py [hunk 3]
```diff
@@ -26,7 +26,7 @@
 import sys
 from halo import Halo
 import go_questionary
-
+from utils import personalize
 __version__ = "0.0.11"  # current version
```
Comment (ShishirPatil) [inline go_cli.py]: bump version for a new push

Comment (ShishirPatil) [reply go_cli.py]: Should also change the setup.py to the new version.

File: go_cli.py [hunk 4]
```diff
@@ -54,6 +58,13 @@
 def generate_random_uid():
     return str(uuid.uuid4())
 
+def load_config():
```
Comment (ShishirPatil) [inline go_cli.py:64]: We need to handle this carefully. 
1. All the user info including user id should be stored in this config. 
2. But at the same time, we should be backward compatible. If a user updates this system, then we should not assign them a new user-id and carry forward their previous user-id.

**Comment by ShishirPatil** [comment]
Given the large scope of this PR, and the fact that we need to update the backend, I'd think we should merge #46 potentially before this.

File: utils.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
+import os
+from presidio_analyzer import AnalyzerEngine, PatternRecognizer
```
Comment (ShishirPatil) [inline utils.py]: Don't you also need to include this in the requirements `presidio_analyzer`

Comment (arthbohra) [reply utils.py]: Where would I access the file for this?

**Pull request closed without merging**